packages/editor/src/components/block-removal-warnings/index.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * WordPress dependencies
+ */
+
+import { __ } from '@wordpress/i18n';
+import { privateApis as blockEditorPrivateApis } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+
+const { BlockRemovalWarningModal } = unlock( blockEditorPrivateApis );
+
+// Prevent accidental removal of certain blocks, asking the user for confirmation first.
+const blockRemovalRules = {
+	'core/query': {
+		message: __(
+			'The Query Loop block displays a list of posts or pages, so removing it will prevent that content from displaying.'
+		),
+	},
```

Comment:
Are we showing this message for the removal of all query blocks?

Reply:
In trunk it is only shown on every deletion in the site editor for query and post content - which makes sense in the context of a template, but if we are adding these rules via the `editor` package I think we need to somehow limit their application to posts/pages - it doesn't seem like a warning is needed in the context of post and pages for these blocks and in fact would be confusing/annoying?

Reply:
have added a `postTypes` option to the rules to allow setting which post types to show the rules for.